jenkins/spark-tests.sh

```diff
@@ -93,6 +93,12 @@ pushd $RAPIDS_INT_TESTS_HOME
 TEST_TYPE="nightly"
 spark-submit $BASE_SPARK_SUBMIT_ARGS --jars $RAPIDS_TEST_JAR ./runtests.py -v -rfExXs --std_input_path="$WORKSPACE/integration_tests/src/test/resources/" --test_type=$TEST_TYPE
 spark-submit $BASE_SPARK_SUBMIT_ARGS $CUDF_UDF_TEST_ARGS --jars $RAPIDS_TEST_JAR ./runtests.py -m "cudf_udf" -v -rfExXs --cudf_udf --test_type=$TEST_TYPE
+#only run cache tests with our serializer in nightly test for Spark version >= 3.1.1
+if [ "$(printf '%s\n' "3.1.1" "$SPARK_VER" | sort -V | head -n1)" = "3.1.1" ]; then
```

Comment:
1. Why exactly is it that we should not run the cache tests unless the version is 3.1.1 or later?
2. Why is it OK to run them in the pre-commit tests but not in the nightly tests?
3. Why not change the test itself to skip things if the versions are wrong/it is configured incorrectly.  We already check versions in test https://github.com/NVIDIA/spark-rapids/blob/6274fadf2393ebf7d47ead7e322e672776cbefc1/integration_tests/src/main/python/cache_test.py#L118-L119

Reply:
1. Because we are already running the `cache_test.py` as part of the integration tests, whats the point of running it again.
2. Its completely OK and we are running `cache_test.py` as part of the precommit. May be we need another PR to address the precommit test so we run the Serializer test as well. 
3. We actually don't want to skip the tests as we want to run them with the default serializer

I hope I have answered the questions if I have misunderstood please let me know and I will try to understand better

Reply:
Then how exactly does this "Fixed nightly build failure in cache tests"?

We are running a test again, this time with a slightly different configuration and that some how makes the existing test stop failing?

Reply:
Sorry, I forgot to change the title when I changed the description. 

The tests weren't enabled for the serializer, this PR will run them now